tethysapp/temp_precip_trends/controllers/api.py

```diff
@@ -121,17 +131,19 @@ def get_projected_mean_temperature(request):
             catalog = app.get_spatial_dataset_service(app.SET_THREDDS_SDS_NAME, as_engine=True)
             dataset = catalog.datasets[app.get_custom_setting(app.SET_THREDDS_DATASET_NAME)]
             params = request.GET.copy()  # create a mutable copy
-            params['start_time'] = datetime.strptime(params['end_time'], '%Y%m%d') + relativedelta(months=-21)
-            params['end_time'] = datetime.strptime(params['end_time'], '%Y%m%d') + relativedelta(months=-9)
-            time_series = get_data('mean_t2m_c', dataset, params)
+            params['start_time'] = dt.datetime.strptime(params['end_time'], '%Y%m%d') + relativedelta(months=-21)
+            params['end_time'] = dt.datetime.strptime(params['end_time'], '%Y%m%d') + relativedelta(months=-9)
+            # TODO: Implement offset dates
+            time_series = get_data('mean_t2m_c', dataset, params, offset_dates=relativedelta(months=12))
```

Comment:
I'll turn this around and ask why we are passing things in through the param object. I've refactored so that the get_data function takes all the params explicitly. So only the controller is dealing with the param objects.

Reply:
Done